config/README.md

```diff
@@ -166,6 +167,226 @@ fmt.Println(m.World)
+  and caches values in underlying map. It also registers callbacks to track
+  changes in all values it cached, so you can call `cached.Get("something")`
+  and don't worry about latencies much. It is safe for concurrent use by
+  multiple goroutines.
+
+* `MockDynamicProvider` is a mock provider that can be used to test dynamic
+  features, it implements `Provider` interface and lets you to set values
+  to trigger change callbacks.
+
+* Sometimes dynamic providers let you to register only one callback per key.
+  If you want to have multiple keys per callback you can use
+  `NewMultiCallbackProvider(p Provider)` wrapper, that will store a list of
+  all callbacks for each value and call them when a value changes.
+  Caution: it locks provider during callbacks execution, you should try to
+  make this callbacks as fast as possible.
+
+* `NopProvider` is useful for testing because it can be embedded in any type
+  if you are not interested in implementing all Provider methods.
+
+* `NewProviderGroup(name string, providers ...Provider)` groups providers in one.
+  Lookups for values are determined by the order providers passed:
+  `NewProviderGroup("global", provider1, provider2)`, first `provider1` will be
+  checked and if there is no value, it will return `provider2.Get()`.
+
+* `NewStaticProvider(data interface{})` is very a useful wrapper for testing,
+  you can pass custom maps and use them as configs instead of loading them
+  from files.
+
+## Loading Configuration
+
+Load process is controlled by `config.Loader`. If a service doesn't specify a
+config provider, manager is going to use a provider returned by
+`config.DefaultLoader.Load()`.
+
+The default loader will load static providers first:
+
+* YAML provider will look for `base.yaml` and `${environment}.yaml` files in
+  current folder and then in `./config` folder. You can override folders
+  to look for these files with `Loader.SetDirs()`.
+  To override files names use `Loader.SetFiles()`.
+
+* Command line provider will look for `--roles` argument to specify service
+  roles. You can introduce/override config values by adding new flags to
+  `pflags.CommandLine` set before building a service.
+
+You can add more static providers on top of mentioned above with
+`RegisterProviders()` function:
+
+```go
+config.DefaultLoader.RegisterProviders(
+  func() Provider, error {
+    return config.NewStaticProvider(map[string]int{"1+2": 3})
+  }
+)
+```
+
+After static providers are loaded, they are used to create dynamic providers.
+You can add new ones in the loader with `RegisterDynamicProviders()` call as well.
+
+In the end all providers are grouped together using
+`NewProviderGroup("global", staticProviders, dynamicProviders)` and returned to service.
+
+If all you want is just a config, there is no need to build a service, you can use
+`config.DefaultLoader.Load()` and get exactly the same config.
+
+Loader type is very customizable and lets you write parallel tests easily: if you
```

Comment:
- The loader type is customizable, letting you write parallel tests easily. If you